src/main/java/rx/internal/util/SubscriptionList.java

```diff
@@ -34,7 +29,6 @@
 
     private LinkedList<Subscription> subscriptions;
     private volatile boolean unsubscribed;
-    private final ReentrantLock lock = new ReentrantLock();
```

Comment:
So is the performance difference related to how the JVM treats `synchronized` vs a `ReentrantLock`?